### PR TITLE
Enable service level readme generation for daily docs.

### PR DESCRIPTION
**Pull request by sima-zhu**
# Description

This is the work of Docs.Ms service readme automation. One prerequisite PR in common https://github.com/Azure/azure-sdk-tools/pull/3578.

Testing in docs.ms: 
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1706901&view=results

Review site: https://review.docs.microsoft.com/en-us/java/api/overview/azure/?view=azure-java-stable&branch=daily%2F2022-05-02-ci-succeeded

### CODE REVIEW

File: eng/scripts/docs/Docs-ToC.ps1 [hunk 1]
```diff
@@ -20,6 +45,11 @@ function Get-java-DocsMsTocData($packageMetadata, $docRepoLocation) {
         $packageLevelReadmeName = $readmeMetadata.DocsMsReadMeName
     }
 
+    return $packageLevelReadmeName
+}
+
+function Get-java-DocsMsTocData($packageMetadata, $docRepoLocation) {
+    $packageLevelReadmeName = Get-java-PackageLevelReadme -packageMetadata $packageMetadata
```
Comment (danieljurek) [inline eng/scripts/docs/Docs-ToC.ps1:52]:
```suggestion
    $packageLevelReadmeName = &$GetPackageLevelReadmeFn -packageMetadata $packageMetadata
```

Is `Get-java-PackageLevelReadme` defined somewhere else?

**Review by danieljurek — COMMENTED** [review]
Looks good so far. However it looks like one of the files is capturing too much in the metadata part of the markdown https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1690507&view=logs&j=dc056dfc-c0cf-5958-c8c4-8da4f91a3739&t=a6454728-bccf-57f0-079a-ebcde92e8213&l=33

File: eng/scripts/docs/Docs-ToC.ps1 (hunk 1 above)
Comment (sima-zhu) [reply eng/scripts/docs/Docs-ToC.ps1:52]: Talked offline. Will use common helper function for the implementation.

File: eng/scripts/Language-Settings.ps1 [hunk 2]
```diff
@@ -6,6 +6,8 @@ $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/main/_data/rel
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=java%2F&delimiter=%2F"
 $CampaignTag = Resolve-Path (Join-Path -Path $PSScriptRoot -ChildPath "../repo-docs/ga_tag.html")
 $packageDownloadUrl = "https://repo1.maven.org/maven2"
+$GithubUri = "https://github.com/Azure/azure-sdk-for-java"
+$RepositoryUri = "https://repo1.maven.org/maven2"
```
[the remark this answers is not in the document]
Comment (sima-zhu) [reply eng/scripts/Language-Settings.ps1:10]: I can perform another PR for rename. I did not see this at the time I checked in the tool repo.

[the remark this answers is not in the document]
Comment (weshaggard) [reply eng/scripts/Language-Settings.ps1:10]: Given this is only used in your one function I wouldn't even have a global variable for it I would just include it local to your function.

**Review by weshaggard — APPROVED** [review]
Left a few comments for clean-up but otherwise seems reasonable.

File: eng/scripts/Language-Settings.ps1 [hunk 3]
```diff
@@ -6,6 +6,8 @@ $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/main/_data/rel
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=java%2F&delimiter=%2F"
 $CampaignTag = Resolve-Path (Join-Path -Path $PSScriptRoot -ChildPath "../repo-docs/ga_tag.html")
 $packageDownloadUrl = "https://repo1.maven.org/maven2"
+$GithubUri = "https://github.com/Azure/azure-sdk-for-java"
```
[the remark this answers is not in the document]
Comment (sima-zhu) [reply eng/scripts/Language-Settings.ps1:9]: Yes, our table include Github source code link

File: eng/scripts/Language-Settings.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (sima-zhu) [reply eng/scripts/Language-Settings.ps1]: My first testing is in JS repo which I did not realize Java has existing var to reuse.
Two options here:
1. Use `PackageRepositoryUrl` anyway
2. Made change to tool `Service-level-readme-automation.ps1`, which default the link to empty(or `$packageDownloadUri/$pkg.Package`) and use the var `$packageDownloadUri` in other languages as well.
@weshaggard Could you share your thoughts?

File: eng/scripts/Language-Settings.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (weshaggard) [reply eng/scripts/Language-Settings.ps1:9]: Ok. For other places were we need to build a URL to the repo we generally use the Language property and build it commonly.

File: eng/scripts/Language-Settings.ps1 [hunk 4]
```diff
@@ -6,6 +6,8 @@ $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/main/_data/rel
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=java%2F&delimiter=%2F"
 $CampaignTag = Resolve-Path (Join-Path -Path $PSScriptRoot -ChildPath "../repo-docs/ga_tag.html")
 $packageDownloadUrl = "https://repo1.maven.org/maven2"
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:8]: Let's join this with PackageRepositoryUri and just update the current usages of packageDownloadUrl to use this new variable.

**Pull request merged**